tests/exec_unit_tests.rs

```diff
@@ -336,3 +347,334 @@ fn multi_set_exit_code_if_command_fails() {
     matcher.finished_dir(Path::new("test_data/simple"), &mut matcher_io);
     assert!(matcher_io.exit_code() == 1);
 }
+
+// -- tests for checking path integrity
+// SAFETY:
+// The only use of unsafe was for the std::env::set_var function.
+// Safety is guaranteed because all tests that directly depend
```

Comment:
Not really true, `set_var` is unsafe in a multi-threaded program unless you can guarantee that no other thread is accessing the environment at all (not just `PATH`).

For unit tests, I think we should just allow passing the `PATH` in.  For integration tests, we can spawn `find` with a new environment without calling `env::set_var()`.